### PR TITLE
Goodbye PhantomJS

### LINKED ISSUE #2150: Switch to headless chromium
We currently use the phantomjs headless browser in capybara through poltergeist. Development of phantomjs has stoppped so we should replace it with something maintained. People move to headless chromium or firefox. We should too...

### PR DESCRIPTION
**Pull request by bear454**
* Drop phantomjs; use selenium to run Chrome or Firefox, visible or headless. (Headless Chrome by default - including features testing in Travis.) Resolves #2150 .

### CODE REVIEW

**Comment by bear454** [comment]
@hennevogel please review...

File: dotenv.example [hunk 1]
```diff
@@ -115,3 +115,12 @@
 # Speeds up turn around times of tests
 # OSEM_FACTORY_LINT=false
 
+# How should browser tests be performed?
```
Comment (hennevogel) [inline dotenv.example]: Why does this need to be configurable?

Comment (bear454) [reply dotenv.example]: _I_ want to test on Chrome _and_ Firefox, but I don't see the need to run the whole suite on both browsers all the time, so, it's configurable. In this way, I can either

```
rspec spec/features
````

to just run the suite... or 

```
OSEM_TEST_DRIVER=firefox rspec spec/features
```

to, for example, see it in Firefox. Makes me happy.

Comment (hennevogel) [reply dotenv.example]: >  Makes me happy.

If that's the case, ship it! :heart_eyes:

**Comment by bear454** [comment]
@hennevogel : so the database issue is mysql only... are you okay with testing on sqlite instead?

File: .travis.yml [hunk 2]
```diff
@@ -23,17 +22,12 @@ notifications:
...
 script:
   - "./travis_script.sh $TEST_SUITE"
 env:
   global:
-    - OSEM_DB_ADAPTER=mysql2
-    - OSEM_DB_USER=root
-    - OSEM_DB_PASSWORD=''
-    - OSEM_DB_HOST=''
-    - OSEM_DB_PORT=''
+    - OSEM_DB_ADAPTER=sqlite3
```
Comment (hennevogel) [inline .travis.yml]: Phew this is very far away from any production instance I know. Might be a temporary fix, but not a solution.

Comment (bear454) [reply .travis.yml]: Well, that's supposed to be the point of ActiveRecord, that the database behind the ORM doesn't matter. If we're worried about full-stack production testing, we'd need to run parallel on MySQL & Postgres. IMO the only thing interim about this as a test db, is that we're not running in-memory. Yet.

Comment (bear454) [reply .travis.yml]: Not to be argumentative, but just to be clear: the `osem` code didn't change, but now the tests pass. This is a pretty clear indicator that the interaction of capybara+database_cleaner+mysql is the culprit for our flaky tests, not the test suite itself.

Comment (hennevogel) [reply .travis.yml]: > capybara+database_cleaner+mysql is the culprit for our flaky tests,

Nobody debates this.

Comment (hennevogel) [reply .travis.yml]: Define `good` :wink: I fear we are going to stay with this a long time. Because it's so convenient...

**Comment by bear454** [comment]
I'm going with sqlite for now. I'll file an issue for moving travis to a "production-grade" database, when the test suite can be run reliably.

**Comment by hennevogel** [comment]
:+1:

**Pull request merged**